### PR TITLE
server: set network rate for additional public IPs

### LINKED ISSUE #5669: VMware: Migrate VR with additional Public IPs failed
Steps to reproduce the issue

(1) create 2 public Ip ranges
(2) create an isolated network, and a vm
(3) acquire a public IP in new ip range (not same as Source NAT), create port forwarding rules to the vm. VR will have two public NICs (source NAT IP, and new public IP).
```
# esxcli network vm port list -w 178877 |grep Portgroup

   Portgroup: cloud.private.untagged.0.1-vSwitch0
   Portgroup: cloud.public.51.200.1-vSwitch1
   Portgroup: cloud.guest.1206.200.1-vSwitch1
   Portgroup: cloud.public.670.200.1-vSwitch1
```

(4) choose a destination host, remove the port group `cloud.public.670.200.1-vSwitch1` on it (this normally happens on vm migration across clusters while the port group does not exist on destination host).

```
# esxcli network vswitch standard portgroup remove -p cloud.public.670.200.1-vSwitch1 -v vSwitch1
```

(5) migrate VR  to destination host, failed with error
```
2021-11-08 08:58:41,867 DEBUG [c.c.a.t.Request] (DirectAgent-129:ctx-1ca672fa) (logid:36218135) Seq 1-1048212813270500266: Processing:  { Ans: , MgmtId: 32986087031327, via: 1(10.0.33.238), Ver: v1, Flags: 10, [{"com.cloud.agent.api.MigrateAnswer":{"result":"false","details":"MigrationCommand failed due to Exception: java.lang.RuntimeException
Message: Currently connected network interface 'Network adapter 5' uses network 'cloud.public.670.200.1-vSwitch1', which is not accessible.
","wait":"0","bypassHostMaintenance":"false"}}] }
```

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Vmware
~~~

##### CLOUDSTACK VERSION

~~~
4.16.0.0-RC3
~~~

##### CONFIGURATION

vmware
basic networkking

##### SUMMARY

see title

##### STEPS TO REPRODUCE

~~~
see description
~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #5669 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Review by GutoVeronezi — APPROVED** [review]
LGTM, I didn't test it though.

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test centos7 vmware-70u1 keepEnv

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java [hunk 1]
```diff
@@ -2069,7 +2068,8 @@ public void prepareAllNicsForMigration(final VirtualMachineProfile vm, final Dep
                     profile.setIsolationUri(Networks.IsolationType.Vlan.toUri(publicIp.getVlanTag()));
                     profile.setSecurityGroupEnabled(_networkModel.isSecurityGroupSupportedInNetwork(network));
                     profile.setName(_networkModel.getNetworkTag(vm.getHypervisorType(), network));
-                    profile.setNetworId(network.getId());
+                    profile.setNetworkRate(_networkModel.getNetworkRate(network.getId(), vm.getId()));
```
Comment (DaanHoogland) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:2071]: will the network id not be needed anymore?

Comment (weizhouapache) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:2071]: @DaanHoogland 
it is still needed. see line 2072
I changed setNetworId to setNetworkId

**Comment by weizhouapache** [comment]
workaround for #5669 : create port group manually on destination host

for example
```
[root@ref-trl-2125-v-M7-wei-zhou-esxi1:~] esxcli network vswitch standard portgroup add -p clo
ud.public.668.200.1-vSwitch1 -v vSwitch1

[root@ref-trl-2125-v-M7-wei-zhou-esxi1:~] esxcli network vswitch standard portgroup set -p clo
ud.public.668.200.1-vSwitch1 --vlan-id 668
```

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Pull request merged**